File: all-in-one-sample/src/main/java/org/arcore/samples/SystemInfoFragment.java
package org.arcore.samples;

import android.annotation.SuppressLint;
import android.os.Build;
import android.os.Bundle;
import android.preference.PreferenceFragment;


public class SystemInfoFragment extends PreferenceFragment {

    private static final String PREFERENCE_ = "preference_";

    @SuppressLint("HardwareIds")
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.prefs);

        //		Build.DEVICE
        //		Build.ID
        //		Build.MANUFACTURER
        //		Build.MODEL
        //		Build.PRODUCT
        //		Build.TAGS
        //		Build.TYPE
        //		Build.USER

        findPreference(PREFERENCE_ + "BOARD").setSummary(Build.BOARD);
        findPreference(PREFERENCE_ + "BRAND").setSummary(Build.BRAND);
        findPreference(PREFERENCE_ + "CPU_ABI").setSummary(Build.SUPPORTED_ABIS[0]);
        findPreference(PREFERENCE_ + "DISPLAY").setSummary(Build.DISPLAY);
        findPreference(PREFERENCE_ + "USER").setSummary(Build.USER);
        findPreference(PREFERENCE_ + "OpenCV_Version").setSummary(BuildConfig.VERSION_NAME);

    }

}

File: all-in-one-sample/src/main/java/org/arcore/samples/SystemInfoFragment.kt
package org.arcore.samples

import android.os.Build
import android.os.Bundle
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat

class SystemInfoFragment : PreferenceFragmentCompat() {

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.prefs)
        //		Build.DEVICE
        //		Build.ID
        //		Build.MANUFACTURER
        //		Build.MODEL
        //		Build.PRODUCT
        //		Build.TAGS
        //		Build.TYPE
        //		Build.USER
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "BOARD")?.summary = Build.BOARD
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "BOARD")?.summary = Build.BOARD
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "BRAND")?.summary = Build.BRAND
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "CPU_ABI")?.summary = Build.SUPPORTED_ABIS[0]
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "DISPLAY")?.summary = Build.DISPLAY
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "USER")?.summary = Build.USER
        preferenceManager.findPreference<Preference>(PREFERENCE_ + "ARCore_Version")?.summary = BuildConfig.VERSION_NAME
    }

    companion object {
        private const val PREFERENCE_ = "preference_"
    }
}